### PR TITLE
Migrate from glog to ESP's logging

### LINKED ISSUE #121: Proposal: replace glog with Corrade::Utility::Debug
I'm seeing a lot of code like

```cpp
LOG(INFO) << "User provided object inertia " 
          << bInertia.x() << " " << bInertia.y() << " " << bInertia.z();
```

or a `switch` that looks like 

```cpp
      case Magnum::MeshPrimitive::Lines:
        LOG(ERROR) << "Invalid primitive: Lines";
        break;
      case Magnum::MeshPrimitive::Points:
        LOG(ERROR) << "Invalid primitive: Points";
        break;
      case Magnum::MeshPrimitive::LineLoop:
        LOG(ERROR) << "Invalid primitive Line loop";
        break;
      ...
```

It's making me sad, because it wastes programmer time -- logging / printf debugging should not involve any mental overhead of "how the heck do I print this variable". The [Utility::Debug](https://doc.magnum.graphics/corrade/classCorrade_1_1Utility_1_1Debug.html) class in Magnum can do the same as just:

```cpp
Debug{} << "User provided object inertia" << bInertia;
```

and it can also print *any* container, almost all Magnum enum values (as names, not numbers, of course), it has output redirection (and muting), debug/warning/error levels, colored output if you want to get fancy and it can [print images](https://blog.magnum.graphics/backstage/multidimensional-strided-array-views/#enter-strided-array-views) as well.

I never used glog, but looking at the project page there doesn't even seem to be any real docs for it, apart from a [raw HTML file](https://github.com/google/glog/blob/master/doc/glog.html). How well maintained that project is? For the WebGL build in #53 we'd need to use something else anyway, so I would propose switching away from it completely.

My question is -- is there any glog feature that `Utility::Debug` doesn't have and you need it? One thing I'm aware of is that it doesn't prefix the output with file/line by default, but there's a (not yet merged) proposal that adds it as an opt-in feature on supported compilers: https://github.com/mosra/corrade/pull/55

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

Migration from glog to ESP's logging from #1382.  I also made some thing that previously loud logs on `LOG(INFO)` (now `ESP_DEBUG()`) be verbose logs, so `ESP_VERY_VERBOSE()`.  

The two keys ones are
* URDF logging.  That use to have it's own compile-time flag, but how you can set urdf and physics to VeryVerbose and get those, i.e. use `export HABITAT_SIM_LOG=physics,urdf=VeryVerbose`
* The logging for scene dataset ingestion.  That was just on INFO but is very noisy for large datasets (like hm3d).  That is partially in metadata and partially in core, so `export HABITAT_SIM_LOG=core,metadata=VeryVerbose` re-enables those.

Most of this migration was done with a script: https://gist.github.com/erikwijmans/499c92fb4eb836dba961b1f9184561b7#file-replace_glog-py

That script is useful both for looking at this PR and for migrating any code in a PR.

Closes #121 

## How Has This Been Tested

Things compile, tests pass, logging looks nice.

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by mosra — COMMENTED** [review]
I didn't go through everything, by far, but the worst that can happen is that some messages will have some extra spaces, I think?

I'd say everyone quickly checks messages of the subsystem they are an "owner" of and then this is good to go? :)

File: src/esp/core/logging.h [hunk 1]
```diff
@@ -80,14 +90,18 @@ inline esp::logging::Subsystem espLoggingSubsystem() {
...
-ADD_SUBSYSTEM_FN(metadata);
-ADD_SUBSYSTEM_FN(geo);
-ADD_SUBSYSTEM_FN(io);
+ESP_ADD_SUBSYSTEM_FN(gfx);
+ESP_ADD_SUBSYSTEM_FN(scene);
+ESP_ADD_SUBSYSTEM_FN(sim);
+ESP_ADD_SUBSYSTEM_FN(physics);
+ESP_ADD_SUBSYSTEM_FN(nav);
+ESP_ADD_SUBSYSTEM_FN(metadata);
+ESP_ADD_SUBSYSTEM_FN(geo);
```
[the remark this answers is not in the document]
Comment (erikwijmans) [reply src/esp/core/logging.h:99]: Yeah, they aren't needed.   Force of habit :)

File: src/esp/core/logging.h [hunk 2]
```diff
@@ -331,56 +345,4 @@ class LogMessageVoidify {
...
-
-#define ASSERT(x, ...)                                                   \
-  do {                                                                   \
-    if (!(x)) {                                                          \
-      Corrade::Utility::Error{Corrade::Utility::Debug::Flag::NoSpace}    \
-          << "Assert failed: " #x << "," << __FILE__ << ":" << __LINE__; \
-      exit(-1);                                                          \
-    }                                                                    \
-  } while (false)
-
```
[the remark this answers is not in the document]
Comment (erikwijmans) [reply src/esp/core/logging.h:385]: Removing this in favor of `CORRADE_INTERNAL_ASSERT`/`CORRADE_ASSERT`/`ESP_CHECK`.

File: src/esp/core/logging.cpp [hunk 3]
```diff
@@ -59,9 +59,6 @@ LoggingLevel levelFromName(const Corrade::Containers::StringView name) {
 }
 
 namespace {
-#ifdef CORRADE_BUILD_MULTITHREADED
-CORRADE_THREAD_LOCAL
-#endif
```
[the remark this answers is not in the document]
Comment (mosra) [reply src/esp/core/logging.cpp:64]: ~~Quick thought -- have a thread_local mutable instance and expose a non-thread-local const pointer?~~ Eh, no, that won't prevent races either, this way the originating thread can still modify the state while other threads read it.

I suppose a way would be to make the `LoggingContext` completely immutable (without `processEnvString()` or `reinitializeFromEnv()`), so reparsing the environment would mean recreating the whole instance, which, due to the RAII nature of it, is rather unlikely to happen in a non-thread-safe way? To me that sounds like a good thing to do in any case, immutable means less accidents.

[the remark this answers is not in the document]
Comment (erikwijmans) [reply src/esp/core/logging.cpp:64]: My rationale for making things mutable and having `reinitializeFromEnv()` was for python.  Since the logging context is a module level global there, you have to set the environment variable before importing habitat_sim.  Not a huge limitation but it seemed nice to let users set that in code if they so desired and just re-parse the environment variable on simulator spin-up.

File: src/esp/metadata/managers/StageAttributesManager.cpp [hunk 4]
```diff
@@ -202,12 +202,20 @@ StageAttributes::ptr StageAttributesManager::initNewObjectInternal(
...
+      const std::string proposed = io::changeExtension(attributesHandle, ext);
+      if (Cr::Utility::Directory::exists(proposed)) {
+        houseFilename = proposed;
+        break;
+      }
+    }
 
     if (!Corrade::Utility::Directory::exists(houseFilename)) {
       houseFilename = io::changeExtension(attributesHandle, ".scn");
     }
```
Comment (jturner65) [inline src/esp/metadata/managers/StageAttributesManager.cpp:216]: This should probably be removed - it seems to be covered by the newly added code.

File: src/esp/physics/bullet/BulletArticulatedObject.cpp [hunk 5]
```diff
@@ -30,7 +30,7 @@ static void setRotationScalingFromBulletTransform(const btTransform& trans,
 ///////////////////////////////////
 
 BulletArticulatedObject::~BulletArticulatedObject() {
-  // Corrade::Utility::Debug() << "deconstructing ~BulletArticulatedObject";
+  // ESP_DEBUG()     << "deconstructing ~BulletArticulatedObject";
```
Comment (jturner65) [inline src/esp/physics/bullet/BulletArticulatedObject.cpp:33]: Remove?

**Review by jturner65 — APPROVED** [review]
LGTM.  Two comments about some potential removals, but otherwise, looks great.

File: src/esp/metadata/managers/StageAttributesManager.cpp (hunk 4 above)
Comment (erikwijmans) [reply src/esp/metadata/managers/StageAttributesManager.cpp:216]: Whoops, I didn't think that ended up in this PR.

**Comment by erikwijmans** [comment]
I misunderstood the source of the `ASSERT_*` macros (those are gtest not glog).  Removed glog for real!

File: src/esp/core/logging.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (mosra) [reply src/esp/core/logging.cpp:64]: Okay, since Python doesn't really have a concept of multithreading anyway, what about exposing `reinitializeFromEnv()` as a Python-only thing, which would be a lambda that internally recreates the `LoggingContext` instance?

File: src/esp/metadata/managers/StageAttributesManager.cpp [hunk 6]
```diff
@@ -202,14 +202,7 @@ StageAttributes::ptr StageAttributesManager::initNewObjectInternal(
...
     // Build default semantic descriptor file name
-    std::string houseFilename = "";
-    for (const auto& ext : {".house", ".scn"}) {
-      const std::string proposed = io::changeExtension(attributesHandle, ext);
-      if (Cr::Utility::Directory::exists(proposed)) {
-        houseFilename = proposed;
-        break;
-      }
-    }
+    std::string houseFilename = io::changeExtension(attributesHandle, ".house");
```
Comment (mosra) [inline src/esp/metadata/managers/StageAttributesManager.cpp]: This change seems completely unrelated? :thinking:

File: src/esp/core/logging.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (erikwijmans) [reply src/esp/core/logging.cpp:64]: I like this idea.  I'll do that.

**Review by mosra — APPROVED** [review]
Looking great from my side :)

**Review by aclegg3 — APPROVED** [review]
Roughly looked over everything with more depth on the URDF area. Seems great, thanks for implementing this! Haven't run it locally, but I could do so if you think we need more testing. Let me know.
